tests/src/test_helpers/ssl_helpers.c

```diff
@@ -115,8 +115,7 @@ void mbedtls_test_ssl_buffer_init(mbedtls_test_ssl_buffer *buf)
 int mbedtls_test_ssl_buffer_setup(mbedtls_test_ssl_buffer *buf,
                                   size_t capacity)
 {
-    buf->buffer = (unsigned char *) mbedtls_calloc(capacity,
-                                                   sizeof(unsigned char));
+    buf->buffer = mbedtls_calloc(capacity, sizeof(unsigned char));
```

Comment:
Since we're changing this line anyway, can we replace the redundant `sizeof(unsigned char)` with `1` please? No need having to read extra text just for that.

Reply:
Bigger issue across the lib is that there is sizeof(variable_type) and not sizeof(*variable_name), that is by far better and less bug prone. No need to repeat ourselves in the code.

See accepted answer here: https://stackoverflow.com/questions/605845/should-i-cast-the-result-of-malloc-in-c

Another problem is that code is inconsistent. Sometimes there is assignment of malloc inside if statement, other time it is before....